src/libtriton/engines/symbolic/symbolicEngine.cpp

```diff
@@ -640,10 +617,10 @@ namespace triton {
 
 
       /* Adds a new symbolic variable */
-      SymbolicVariable* SymbolicEngine::newSymbolicVariable(triton::engines::symbolic::symkind_e kind, triton::uint64 kindValue, triton::uint32 size, const std::string& comment) {
+      SharedSymbolicVariable SymbolicEngine::newSymbolicVariable(triton::engines::symbolic::symkind_e kind, triton::uint64 kindValue, triton::uint32 size, const std::string& comment) {
         triton::usize uniqueId = this->getUniqueSymVarId();
-        SymbolicVariable* symVar = new(std::nothrow) SymbolicVariable(kind, kindValue, uniqueId, size, comment);
 
+        SharedSymbolicVariable symVar = std::make_shared<SymbolicVariable>(kind, kindValue, uniqueId, size, comment);
```

Comment:
Ici, tu devrais retourner "this->symbolicVariables[uniqueId]"  et non pas symVar, ça te permettrait de retourner un "const SharedSymbolicVariable&" au lieu d'un SharedSymbolicVariable ce que t'évite quelques incref/decref. Aussi il y a des fois ou tu appeles cette fonction et prend le résultat dans un "const SharedSymbolicVariable&" ce qui est mal parce que ton objet est ici détruit (pas la Variable mais le shared_ptr) et tu te retrouves avec un use after free. Faire la modif de signature réglera aussi ce souci.